### PR TITLE
Upsert processing speed Increments

### PR DESCRIPTION
**Pull request by if0s**
* Added new `Type Of Search` - `External IDs` in `Upsert Object` action
* implemented caching for metadata in `Upsert Object` action (metadata needs to find fields that contain attachment)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.2.5 (June 17, 2022)
+* Added new `Type Of Search` - `External IDs` in `Upsert Object` action
+* implemented caching for metadata in `Upsert Object` action (metadata needs to find fields that contain attachment)
```
Comment (A3a3e1) [inline CHANGELOG.md:3]:
```suggestion
* Implemented caching for metadata in `Upsert Object` action (metadata needs to find fields that contain attachment)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 2.2.5 (June 17, 2022)
+* Added new `Type Of Search` - `External IDs` in `Upsert Object` action
```
Comment (A3a3e1) [inline CHANGELOG.md:2]:
```suggestion
* Added new `Type Of Search` - `External IDs` to `Upsert Object` action
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+## 2.2.5 (June 17, 2022)
```
Comment (EmptyInfinity) [inline CHANGELOG.md:1]:
```suggestion
## 2.3.0 (June 17, 2022)
```

File: README.md [hunk 4]
```diff
@@ -331,5 +334,8 @@ The result of creating or updating an object
 * **success** - Boolean result of creation/update object
 * **errors** - Arrey of errors if they exist
 
+#### Known limitations
+If you add new field to object in Salesforce, you should restart flow to let them appear inside input metadata
```
Comment (A3a3e1) [inline README.md:338]:
```suggestion
If you add a new field to an object in Salesforce, you must restart the flow to re-generate metadata
```

File: component.json [hunk 5]
```diff
@@ -4,7 +4,7 @@
   "docsUrl": "https://github.com/elasticio/salesforce-component-v2",
   "url": "http://www.salesforce.com/",
   "buildType": "docker",
-  "version": "2.2.4",
+  "version": "2.2.5-dev.2",
```
Comment (EmptyInfinity) [inline component.json:7]:
```suggestion
  "version": "2.3.0-dev.1",
```

Comment (if0s) [reply component.json]: done

File: lib/actions/upsert_v2.js [hunk 6]
```diff
@@ -51,69 +50,58 @@ module.exports.getMetaModel = async function getMetaModel(configuration) {
...
+    this.logger.info(`Going to upsert "${cfg.sobject}" object using "${cfg.lookupField}" field`);
+    const extIdField = cfg.lookupField;
+    result = await callJSForceMethod.call(this, { ...cfg, timeOut, extIdField }, 'sobjectUpsert', msg);
+  } else {
+    this.logger.info('Starting to get object for an upsert...');
+    const existingObj = await callJSForceMethod.call(this, { ...cfg, timeOut }, 'sobjectLookup', msg);
+    if (existingObj.length > 1) {
+      throw new Error('Found more than 1 Object');
+    } else if (existingObj.length === 0) {
+      this.logger.info('The object has not existed, starting the creating the new one');
```
Comment (A3a3e1) [inline lib/actions/upsert_v2.js:94]:
```suggestion
      this.logger.info('An object does not exist, creating a new one');
```

**Review by A3a3e1 — APPROVED** [review]

**Review by EmptyInfinity — APPROVED** [review]

**Pull request merged**